backend/utils/passport-setup.js

```diff
@@ -19,20 +24,31 @@ passport.use(
       callbackURL: `http://localhost:5000/api/auth/login/callback`
     },
     function(accessToken, refreshToken, profile, cb) {
-      cb(null, "Hello");
+      // find the user in the database based on their facebook id
+      User.findOne({ oauthId: profile.id }, function(err, user) {
+        if (err) {
+          return cb(err);
+        }
+
+        if (user) {
+          console.log("found");
+          return cb(null, user);
+        }
+        new User({
+          firstName: profile.name.givenName,
+          lastName: profile.name.familyName,
+          oauthId: profile.id,
+          propicUrl: profile.photos[0].value,
+          isApproved: false,
+          role: "VOLUNTEER",
+          location: "NORTH"
+        })
+          .save()
+          .then(user => {
```

Comment:
We can actually use async/await syntax here:

`const newUser = await new User({...}).save()`